src/index/CompressedBlockPrefiltering.cpp

```diff
@@ -270,22 +329,98 @@ std::vector<BlockMetadata> LogicalExpression<Operation>::evaluateImpl(
+bool LogicalExpression<Operation>::operator==(
+    const PrefilterExpression& other) const {
+  const auto* otherlogical =
+      dynamic_cast<const LogicalExpression<Operation>*>(&other);
+  if (!otherlogical) {
+    return false;
+  }
+  return *child1_ == *otherlogical->child1_ &&
+         *child2_ == *otherlogical->child2_;
+};
+
+//______________________________________________________________________________
+template <LogicalOperator Operation>
+std::unique_ptr<PrefilterExpression> LogicalExpression<Operation>::clone()
+    const {
+  return std::make_unique<LogicalExpression<Operation>>(child1_->clone(),
+                                                        child2_->clone());
+};
+
+//______________________________________________________________________________
+template <LogicalOperator Operation>
+std::string LogicalExpression<Operation>::info(size_t depth) const {
+  std::string child1Info =
+      depth < maxInfoRecursion ? child1_->info(depth + 1) : "MAX_DEPTH";
+  std::string child2Info =
+      depth < maxInfoRecursion ? child2_->info(depth + 1) : "MAX_DEPTH";
+  std::stringstream stream;
+  stream << "Prefilter LogicalExpression<" << getLogicalOpStr(Operation)
+         << ">\n"
+         << "child1 {" << child1Info << "}"
+         << "child2 {" << child2Info << "}" << std::endl;
+  return stream.str();
+};
+
+// SECTION NOT-EXPRESSION
+//______________________________________________________________________________
+std::unique_ptr<PrefilterExpression> NotExpression::logicalComplement() const {
+  // Logically we complement (negate) a NOT here => NOT cancels out.
+  // Therefore, we can simply return the child of the respective NOT
+  // expression after undoing its previous complementation.
+  return child_->logicalComplement();
+};
+
 //______________________________________________________________________________
 std::vector<BlockMetadata> NotExpression::evaluateImpl(
     const std::vector<BlockMetadata>& input, size_t evaluationColumn) const {
   return child_->evaluate(input, evaluationColumn);
 };
 
 //______________________________________________________________________________
-// Necessary instantiation of template specializations
-template class RelationalExpression<CompOp::LT>;
-template class RelationalExpression<CompOp::LE>;
-template class RelationalExpression<CompOp::GE>;
-template class RelationalExpression<CompOp::GT>;
-template class RelationalExpression<CompOp::EQ>;
-template class RelationalExpression<CompOp::NE>;
+bool NotExpression::operator==(const PrefilterExpression& other) const {
+  const auto* otherNotExpression = dynamic_cast<const NotExpression*>(&other);
+  if (!otherNotExpression) {
+    return false;
+  }
+  return *child_ == *otherNotExpression->child_;
+}
+
+//______________________________________________________________________________
+std::unique_ptr<PrefilterExpression> NotExpression::clone() const {
+  return std::make_unique<NotExpression>((child_->clone()), true);
```

Comment:
I understand. the second constructor argument now.
Document it like "should always be `false`, `true` is only needed for the implementation of `clone()`